framework/doc/content/source/meshgenerators/MeshDiagnosticsGenerator.md

```diff
@@ -0,0 +1,136 @@
+- detecting when a node of an element is inside another element
+
+!alert note
+These two checks are heuristics. They can have false negatives if elements overlap but in a way that the centroids
+and nodes remain outside of other elements. This can happen for example with curved side elements overlapping on a side.
+
+## Non-conformality check
+
+Turn this check on with the [!param](/Mesh/MeshDiagnosticsGenerator/examine_non_conformality) parameter.
+
+Non-conformality in MOOSE is generally only supported if it is arising from adaptive mesh refinement created
+by the underlying mesh library. Additional information about the parent/child relationship between coarse and fine
+elements is stored at the element level. However, numerous external meshes, especially arising from simulations
+using adaptive mesh refinement, will present non-conformality. To support them, we usually resort to special numerical treatments,
+such as mortar methods.
+
+Non-conformality is detected by this diagnostic by looking for nodes that are near an element within a tolerance but
+not part of this element's nodes.
+
+!alert note
+If the non-conformality is extremely minor, and the non-conformal nodes are very close to other nodes,
+they can be merged together using a [MeshRepairGenerator.md].
+
+### Non-conformality arising from Adaptive Mesh Refinement
+
+Turn this check on with the [!param](/Mesh/MeshDiagnosticsGenerator/search_for_adaptivity_nonconformality) parameter.
+
+In the special case of meshes where uniform h-refinement was performed, the `MeshDiagnosticsGenerator` can detect
+at the interface between coarse and fine elements that the fine elements could be combined into a coarse element.
+The coarse element, if it were refined with MOOSE uniform h-adaptivity, would be refined into the same fine elements as
+originally present in the mesh.
+
+This check will only detect uniform mesh refinement for triangle and quadrilateral 2D elements and, tetrahedral and hexahedral 3D
+elements. For tetrahedrals, because the refinement pattern depends on the selection of a diagonal inside the coarse element,
+the check only consider the `tip` fine elements on the four vertex of the coarse element.
+
+!alert note
+This check is a heuristic. It will only detect adaptive *uniform* mesh refinement and it will only detect it
+near interfaces between coarse and fine elements. If the entire mesh is refined the same number of times for each
+element, it will not be detected.
+
+## Elements and sides local Jacobian check
+
+The local Jacobian is checked for positivity for every element and sides. Negative Jacobians are a common issue
+with a poor, usually externally generated, mesh.
+
+!alert note
+A Lagrange finite element family of the default order for each element is used for this check, along with a fourth-order
+Gauss quadrature. If you intend to use another finite element family,
+feel free to modify the generator for your needs.
```

Comment:
Ah, I get it now, after looking through the code.  You're using `FE::reinit()` rather than just creating an `FEMap` and `QGauss` and calling the methods at that level by hand, and you have to pick *some* type of FE.  Our fault at the libMesh level for not making the lower-level methods more convenient to use; your code looks fine given that context.

But it doesn't actually matter that your FE type is `LAGRANGE`.  It'd be more efficient to make it a `CONSTANT` `MONOMIAL`, in fact.  And I definitely wouldn't talk about the FE family in the docs here; that's a trick that doesn't affect the results, unlike the quadrature rule.

Reply:
ah I should have known. I started out realizing the quadrature mattered but never thought again about the FE family